src/Confluent.Kafka/Consumer.cs

```diff
@@ -645,13 +659,8 @@ public void Assign(IEnumerable<TopicPartition> partitions)
         public void Unassign()
             => kafkaHandle.Assign(null);
 
-        /// <summary>
-        ///     Manually consume message or triggers events.
-        ///
-        ///     Will invoke events for OnPartitionsAssigned/Revoked,
-        ///     OnOffsetCommit, etc. on the calling thread.
-        /// </summary>
-        public bool Consume(out Message message, int millisecondsTimeout)
+
+        internal bool ConsumeImpl(out Message message, int millisecondsTimeout)
```

Comment:
I would be in favor of leveraging overloads instead of optional parameters, then you can add new signatures without potentially breaking existing clients. e.g. like @tomasdeml mentioned, if you add a second optional param it will break existing clients until they recompile.